code/modules/mob/living/simple_animal/hostile/glockroach.dm

```diff
@@ -0,0 +1,64 @@
+/obj/projectile/glockroachbullet
+	damage = 10 //same damage as a hivebot
+	damage_type = BRUTE
+
+/mob/living/simple_animal/hostile/glockroach //copypasted from cockroach.dm so i could use the shooting code in hostile.dm
```

Comment:
you really should make this a subtype of cockroaches, because there are quite a few things in the game that specifically interact with cockroaches

Reply:
i wanted to make it a child of cockroaches, but the problem then is i couldn't piggyback off the procs in /hostile to allow it to fire

Reply:
oh or the aggro code

Reply:
Maybe you could make ALL cockroaches be a child of hostile?

They ARE all out to get us, y'know.

Reply:
blame the shithole that is NPC code